src/Adapter/Title/CommandHandler/BulkDeleteTitleHandler.php

```diff
@@ -28,32 +28,24 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Title\CommandHandler;
 
-use Gender;
+use PrestaShop\PrestaShop\Adapter\Title\AbstractTitleHandler;
 use PrestaShop\PrestaShop\Core\Domain\Title\Command\BulkDeleteTitleCommand;
 use PrestaShop\PrestaShop\Core\Domain\Title\CommandHandler\BulkDeleteTitleHandlerInterface;
-use PrestaShop\PrestaShop\Core\Domain\Title\Exception\DeleteTitleException;
-use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleNotFoundException;
 
 /**
  * Handles command that bulk delete titles
  */
-class BulkDeleteTitleHandler implements BulkDeleteTitleHandlerInterface
+class BulkDeleteTitleHandler extends AbstractTitleHandler implements BulkDeleteTitleHandlerInterface
 {
     /**
      * {@inheritdoc}
      */
     public function handle(BulkDeleteTitleCommand $command): void
     {
         foreach ($command->getTitleIds() as $titleId) {
-            $title = new Gender($titleId->getValue());
-
-            if (0 >= $title->id) {
-                throw new TitleNotFoundException(sprintf('Unable to find title with id "%d" for deletion', $titleId->getValue()));
-            }
-
-            if (!$title->delete()) {
-                throw DeleteTitleException::createBulkDeleteFailure($titleId);
-            }
+            $this->titleRepository->delete(
+                $this->titleRepository->get($titleId)
+            );
```

Comment:
could have been 
`$this->titleRepository->delete($titleId)` 
and the repository would get the obj model inside.
But its not blocking for me, because recently in some other PR I noticed if the deletion is more complicated (e.g if you would need to check some Title fields before deletion) you would still need to fetch the obj model before so this way would make more sense :man_shrugging:. So idk, both ways are good for me, not sure which is better :smile: